src/plugins/auto/src/plugin.cpp

```diff
@@ -575,91 +575,73 @@ std::list<DeviceInformation> Plugin::get_valid_device(
     if (meta_devices.empty()) {
         OPENVINO_THROW("No available device to select in ", get_device_name());
     }
-
+    bool is_default_list = true;
+    std::list<DeviceInformation> valid_devices;
     std::list<DeviceInformation> CPU;
     std::list<DeviceInformation> dGPU;
     std::list<DeviceInformation> iGPU;
-    std::list<DeviceInformation> MYRIAD;
-    std::list<DeviceInformation> VPU;
+    std::list<DeviceInformation> Others;
+    auto is_supported_model = [this, model_precision](const std::string& device_name) {
+        // Check if candidate device supported the specified model precision.
+        auto capability = get_core()->get_property(device_name, ov::device::capabilities);
+        auto support_model = std::find(capability.begin(), capability.end(), (model_precision));
+        bool is_support_fp16 =
+            model_precision == "FP32" && std::find(capability.begin(), capability.end(), ("FP16")) != capability.end();
+        return support_model != capability.end() || is_support_fp16;
+    };
 
-    for (auto& item : meta_devices) {
-        if (item.device_name.find("CPU") == 0) {
-            CPU.push_back(item);
-            continue;
-        }
-        if (item.device_name.find("MYRIAD") == 0) {
-            MYRIAD.push_back(item);
-            continue;
-        }
-        if (item.device_name.find("VPU") == 0) {
-            VPU.push_back(item);
+    for (auto&& device_info : meta_devices) {
+        if (device_info.device_priority > 0)
+            is_default_list = false;
+        // check if device support this model precision
+        if (!is_supported_model(device_info.device_name) && meta_devices.size() > 1)
             continue;
-        }
-        if (item.device_name.find("GPU") == 0) {
+        if (device_info.device_name.find("CPU") == 0) {
+            CPU.push_back(device_info);
+        } else if (device_info.device_name.find("GPU") == 0) {
             std::string device_type;
             try {
                 // can optimize to typed function when gpu swith to 2.0 api
-                device_type = get_core()->get_property(item.device_name, ov::device::type.name(), {}).as<std::string>();
+                device_type =
+                    get_core()->get_property(device_info.device_name, ov::device::type.name(), {}).as<std::string>();
             } catch (const ov::Exception&) {
-                LOG_DEBUG_TAG("get property :%s for %s failed ", "DEVICE_TYPE", item.device_name.c_str());
+                LOG_DEBUG_TAG("get property :%s for %s failed ", "DEVICE_TYPE", device_info.device_name.c_str());
             }
             if (device_type == "integrated") {
-                iGPU.push_back(item);
+                iGPU.push_back(device_info);
             } else if (device_type == "discrete") {
-                dGPU.push_back(item);
+                dGPU.push_back(device_info);
             } else {
```

Comment:
all these dGPU,iGPU vectors seems no use when is_default_list is false?

Reply:
Yes,  It's convenient to kept this logic that saved the device info into these dGPU, iGPU, CPU vectors during the iteration. Eg, `meta_devices ` may look like as below. 
`{
{"GPU.0",..,0}, //device priority is 0
{"GPU.1",..,0},//device priority is 0
{"GPU.2",..,0},//device priority is 0
{"GPU.3",..,0},//device priority is 0
{"CPU",..,1},//device priority is 1
}`